src/main.py

```diff
@@ -290,7 +288,6 @@ async def main() -> None:
                             latiss_gauge.inc()
                             fan_out_message_list = (
                                 next_visit_message_updated.add_detectors(
-                                    "LATISS",
                                     dataclasses.asdict(next_visit_message_updated),
                                     latiss_active_detectors,
                                 )
```

Comment:
It seems a bit odd to have `add_detectors` be triggered by a match on `salIndex` when we're already assuming that `next_visit_message_updated` has a valid `instrument`. Is this to handle the weird edge case of `upload.py HSC`? If so, it might be a good idea to add a comment to the `match` block.

Reply:
Now that we have a valid `instrument` we probably can refactor this. Indeed the weird case of `upload.py HSC` incomplete detectors will make the refactoring uglier. I'll add a comment.